src/antsibull_docs/process_docs.py

```diff
@@ -63,36 +138,45 @@ def normalize_plugin_info(
 
     errors: list[str] = []
     if plugin_type == "role":
+        role_schema: type[BaseModel] = DOCS_SCHEMAS[
+            plugin_type
+        ]  # type: ignore[attr-defined, assignment]
         try:
-            parsed = DOCS_SCHEMAS[plugin_type].parse_obj(plugin_info)  # type: ignore[attr-defined]
-            return parsed.dict(by_alias=True), errors
-        except v1.ValidationError as e:
-            raise ValueError(str(e))  # pylint:disable=raise-missing-from
+            parsed = role_schema.model_validate(plugin_info)
+            return parsed.model_dump(by_alias=True), errors
+        except p.ValidationError as e:
+            raise ValueError(  # pylint:disable=raise-missing-from
+                _exc_to_string(e, role_schema.__name__)
+            )
 
     new_info: dict[str, t.Any] = {}
     # Note: loop through "doc" before any other keys.
     for field in ("doc", "examples", "return"):
+        schema: type[BaseModel] = DOCS_SCHEMAS[plugin_type][
+            field
+        ]  # type: ignore[index, assignment]
         try:
-            schema = DOCS_SCHEMAS[plugin_type][field]  # type: ignore[index]
-            field_model = schema.parse_obj({field: plugin_info.get(field)})
-        except v1.ValidationError as e:
+            field_model = schema.model_validate({field: plugin_info.get(field)})
+        except p.ValidationError as e:
             if field == "doc":
                 # We can't recover if there's not a doc field
                 # pydantic exceptions are not picklable (probably due to bugs in the pickle module)
                 # so convert it to an exception type which is picklable
-                raise ValueError(str(e))  # pylint:disable=raise-missing-from
+                raise ValueError(  # pylint:disable=raise-missing-from
+                    _exc_to_string(e, schema.__name__)
+                )
 
             # But we can use the default value (some variant of "empty") for everything else
             # Note: We looped through doc first and returned an exception if doc did not normalize
             # so we're able to use it in the error message here.
             errors.append(
                 f'Unable to normalize {new_info["doc"]["name"]}: {field}'
-                f" due to: {str(e)}"
+                f" due to: {_exc_to_string(e, schema.__name__)}"
             )
 
-            field_model = DOCS_SCHEMAS[plugin_type][field].parse_obj({})  # type: ignore[index]
+            field_model = DOCS_SCHEMAS[plugin_type][field].model_validate({})  # type: ignore[index]
```

Comment:
```suggestion
            field_model = DOCS_SCHEMAS[plugin_type][field]()  # type: ignore[index]
```

Reply:
I don't think this is a good idea as long as we use `Field()` with a default value.

Reply:
How does `model_validate()` affect Fields with default values?

Reply:
It should assign the default value from the `Field` instance to them if the field isn't provided the input.

Reply:
It should do the same thing if you call the constructor without parameters, no?

Reply:
I would hope so, but I don't know. After my previous comment I wanted to try it out, and noticed that I can't because all structs that use such fields are not default-constructable (due to some validators). Which also means that all structs where the above construct is used do not use `Field` with a `default` value, so it doesn't really matter and your replacement is fine :)